src/components/manager/projects/ProjectJsonSchemaForm.js

```diff
@@ -0,0 +1,138 @@
+import React, { useCallback } from "react";
+import PropTypes from "prop-types";
+import { Form, Select, Checkbox, Tooltip, Button, message } from "antd";
+import { useDropzone } from "react-dropzone";
+import ReactJson from "react-json-view";
+import Ajv from "ajv";
+
+const ajv = new Ajv({
+    allErrors: true,
+    strict: true,
+});
+
+// Does not actually query over http, the URI is the key to the draft-07 meta-schema
+const validateSchema = ajv.getSchema("http://json-schema.org/draft-07/schema");
+
+const ProjectJsonSchemaForm = ({ style, schemaTypes, initialValues, setFileContent, fileContent, form }) => {
+
+    const onDrop = useCallback((files) => {
+        files.forEach((file) => {
+            const reader = new FileReader();
+            reader.onabort = () => console.error("file reading was aborted");
+            reader.onerror = () => console.error("file reading has failed");
+            reader.onload = () => {
+                const json = JSON.parse(reader.result);
+                if (validateSchema(json)) {
+                    // Validate against draft-07 meta schema
+                    setFileContent(json);
+                } else {
+                    message.error("Selected file is an invalid JSON schema definition.");
+                }
+            };
+            reader.readAsText(file);
+        });
+    }, []);
+
+    const { getRootProps, getInputProps } = useDropzone({
+        onDrop,
+        maxFiles: 1,
+        accept: {
+            "application/json": [".json"]
+        },
+    });
+
+    return (
+        <Form style={style || {}}>
+            <Form.Item label={
+                <Tooltip title="The data type on which this extra_properties schema will be applied">
+                    Schema Type
+                </Tooltip>
+            }>
+                {form.getFieldDecorator("schemaType", {
+                    initialValue: initialValues.schemaType,
+                    rules: [{ required: true }]
+                })(
+                    <Select>
+                        {schemaTypes.map(option => (
+                            <Select.Option key={option} value={option}>
+                                {option}
+                            </Select.Option>
+                        ))}
+                    </Select>
+                )}
+            </Form.Item>
+            <Form.Item label={
+                <Tooltip title="Check to make the extra_properties field required">
```

Comment:
wrap `extra_properties` in <Typography.Text code={true}>...</Typography.Text>

Reply:
see https://github.com/bento-platform/bento_web/pull/228#discussion_r1198223989